web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -69,6 +175,40 @@ export class JobListItemComponent implements OnInit, OnDestroy {
         this.surveyId = id;
       })
     );
+
+    this.subscription.add(
+      this.loiService.getLocationsOfInterest$().pipe(shareReplay()).subscribe(lois => {
+        const jobLois = lois.filter((loi) => {
+          return loi.jobId === this.job?.id;
+        });
+
+        this.surveyService.getActiveSurvey$().subscribe((survey) => {
+          let numOfNoLabelLois = 0;
+          jobLois.forEach((loi) => {
```

Comment:
`jobLois` will be updated at some point after the survey changes, and may not be loaded yet when this is reached. Instead, can we derive streams using standard Rx operators that do what we want here and below?